### PR TITLE
refactor(typescript): Convert TableCell, TableExpandRow, TableExpandedRow, TableRow to tsx

### LINKED ISSUE #12529: Add TypeScript types to TableRow

### PR DESCRIPTION
**Pull request by mbarrer**
Closes #12529, #12522, #12525, #12526

Converts TableCell, TableExpandRow, TableExpandedRow and TableRow to tsx in preparation for the main Table conversion

#### Changelog

**Changed**

- Added JSDoc typings on `wrapComponent.js`
- Converted TableCell.js -> TableCell.tsx
- Converted TableExpandRow.js -> TableExpandRow.tsx
- Converted TableExpandedRow.js -> TableExpandedRow.tsx
- Converted TableRow -> TableRow.tsx

#### Testing / Reviewing

To verify these changes are working, we can check to see if the existing storybook behavior is still working correctly. Should be able to properly import converted components in a TS environment

### CODE REVIEW

File: packages/react/src/tools/wrapComponent.js [hunk 1]
```diff
@@ -10,7 +10,16 @@ import PropTypes from 'prop-types';
 import cx from 'classnames';
 import { usePrefix } from '../internal/usePrefix';
 
+/**
+ * @param {{ name: string, className?: string, type: string }} param0
```
Comment (francinelucca) [inline packages/react/src/tools/wrapComponent.js]:
```suggestion
 * @param {{ name: string, className?: string | Function, type: string }} param0
```

**Review by francinelucca — CHANGES_REQUESTED** [review]
LGTM, one minor change suggestion

File: packages/react/src/tools/wrapComponent.js (hunk 1 above)
Comment (mbarrer) [reply packages/react/src/tools/wrapComponent.js]: Ah ty I missed the function usage. I would prefer something more strict than `Function` though. Would you settle for this instead? 

```
@param {{ name: string, type: string, className?: string | (prefix: string) => string }} props
```

Comment (francinelucca) [reply packages/react/src/tools/wrapComponent.js]: sounds good! though technically the `getClassName` function could return an object with classNames as keys and it'd still work 🤔: 
image

Comment (mbarrer) [reply packages/react/src/tools/wrapComponent.js]: Ah right because most components consume the className with that `classnames` lib. I didnt see any current usages in the code where the getClassName returns an obj, I could change it to this though. might be a bit too wordy for a jsdoc typing and better served when we convert the internal files to ts. Let me know if I should change it

image

**Review by francinelucca — APPROVED** [review]

File: packages/react/src/tools/wrapComponent.js (hunk 1 above)
Comment (francinelucca) [reply packages/react/src/tools/wrapComponent.js]: you're right, doesn't seem like it's being used anywhere returning an object, I'm fine keeping it as-is. thanks!

**Review by tay1orjones — APPROVED** [review]
